packages/wallet/backend/src/grant/service.ts

```diff
@@ -80,6 +80,32 @@ export class GrantService implements IGrantService {
       args.filter = { identifier: { in: identifiers } }
     }
 
-    return await this.rafikiAuthService.listGrantsWithPagination(args)
+    const grants = await this.rafikiAuthService.listGrantsWithPagination(args)
+    this.parseIntervals(grants.grants.edges.map((it) => it.node))
+    return grants
+  }
+
+  private parseIntervals(grants: Grant[]) {
+    for (const grant of grants)
+      for (const accessElement of grant.access)
+        if (accessElement.limits?.interval) {
+          const time = accessElement.limits?.interval.split('/')
+          const duration = moment.duration(time[2])
+          const date = moment(time[1])
+          const months = duration.asMonths()
+          if (months === 1)
+            accessElement.limits!.interval = `every month from ${date.format('ddd')}`
+          else if (months > 1 && months < 12)
+            accessElement.limits!.interval = `every ${months} month from ${date.format('ddd')}`
+          else if (months === 12)
+            accessElement.limits!.interval = `every year from ${date.format('ddd')}`
+          else if (months > 12) {
+            const years = duration.asYears()
+            accessElement.limits!.interval = `every ${years} year from ${date.format('ddd')}`
+          } else if (months < 1 && months > 0) {
+            const days = duration.asDays()
+            accessElement.limits!.interval = `every ${days} days from ${date.format('ddd')}`
+          }
+        }
```

Comment:
How do you feel about an implementation of this method that does not mutate the grants list?
(by using .map)
(This would also imply the above changes for the usage of the function)
```suggestion
  private parseIntervals(grants: Grant[]): Grant[] {
  return grants.map(grant => {
    const access = grant.access.map(accessElement => {
      if (accessElement.limits?.interval) {
        const time = accessElement.limits.interval.split('/')
        const duration = moment.duration(time[2])
        const date = moment(time[1])
        const months = duration.asMonths()

        let interval: string;
        if (months === 1) {
          interval = `every month from ${date.format('ddd')}`;
        } else if (months > 1 && months < 12) {
          interval = `every ${months} month from ${date.format('ddd')}`;
        } else if (months === 12) {
          interval = `every year from ${date.format('ddd')}`;
        } else if (months > 12) {
          const years = duration.asYears();
          interval = `every ${years} year from ${date.format('ddd')}`;
        } else if (months < 1 && months > 0) {
          const days = duration.asDays();
          interval = `every ${days} days from ${date.format('ddd')}`;
        }

        return {
          ...accessElement,
          limits: {
            ...accessElement.limits,
            interval: interval,
          },
        };
      }

      return accessElement;
    });

    return { ...grant, access };
  });
}
```

Reply:
If you find it decreases readability, I guess we could settle for mutating the grants in the parse method as well.
Up to you & the others